Realm/RLMClassInfo.hpp

```diff
@@ -101,6 +116,15 @@ class RLMSchemaInfo {
 
     // Look up by name, throwing if it's not present
     RLMClassInfo& operator[](NSString *name);
+    // Look up by table key, return none if its not present.
+    realm::util::Optional<RLMClassInfo&> operator[](realm::TableKey const& tableKey);
```

Comment:
Optional references have strange and potentially surprising semantics, so outside of generic programming contexts just using a pointer is preferable.

Reply:
This will create issues for `RLMObjectFromObjLink` where the purpose of returning optional is so that if the table key does not exist in RLMSchemaInfo we can create the object dynamically.

Reply:
Returning null and returning none do the exact same thing there.